pulsar-client/src/main/java/org/apache/pulsar/client/impl/PersistentAcknowledgmentsGroupingTracker.java

```diff
@@ -124,7 +124,19 @@ public boolean isDuplicate(MessageId messageId) {
             // Already included in a cumulative ack
             return true;
         } else {
-            return pendingIndividualAcks.contains(MessageIdAdvUtils.discardBatch(messageIdAdv));
+            // If "batchIndexAckEnabled" is false, the batched messages acknowledgment will be traced by
+            // pendingIndividualAcks. So no matter what type the message ID is, check with "pendingIndividualAcks"
+            // first.
+            if (pendingIndividualAcks.contains(MessageIdAdvUtils.discardBatch(messageIdAdv))) {
```

Comment:
`MessageIdAdvUtils` was first introduced since 3.0.0, so if you're going to cherry-pick it to 2.11 or earlier, you need to adjust the code.

Reply:
Sure